src/days-2-4-adv/adv.py

```diff
@@ -32,15 +36,60 @@
 room['narrow'].w_to = room['foyer']
 room['narrow'].n_to = room['treasure']
 room['treasure'].s_to = room['narrow']
+room['treasure'].n_to = room['kitchen']
+room['kitchen'].e_to = room['prison']
+room['prison'].n_to = room['coward']
 
-#
+# Weapons
+weapons = {
+ 
+}
+
+items = {
+    "shovel": Item("shovel", "Use this to dig or hit with"),
+    "sword" : Item("sword", "Use this to chop or slice"),
+    "spells": Item ("spells", "Book of spells"),
+    "bat" : Item("bat", "Use this to knock down or hit with"),
+    "gun" : Item("gun", "Shoot with this"),
+}
 # Main
-#
 
-# Make a new player object that is currently in the 'outside' room.
+rooms = ["outside", "foyer", "overlook",
+         "narrow", "treasure", "kitchen", "prison"]
+
+
+def change_rooms(Player, direction):
+    direction = direction.lower()
+    try:
+        if direction == "n":
+            return Player.room_change(room[Player.roomKey].n_to.name)
+        elif direction == "s":
+            return Player.room_change(room[Player.roomKey].s_to.name)
 
+        elif direction == "e":
+            return Player.room_change(room[Player.roomKey].e_to.name)
+        elif direction == "w":
+            return Player.room_change(room[Player.roomKey].w_to.name)
+    except:
+        return "You can not go in that direction!"
+# end of change_rooms function
+
+
+# Make a new player object that is currently in the 'outside' room.
+jonathan = Player("Jonathan", 'Outside Cave Entrance', "outside")
```

Comment:
Instead of passing a string to the player class, you could pass the actual room. `room['outside']`